cardano-api/src/Cardano/Api/Eras.hs

```diff
@@ -215,6 +218,9 @@ data AnyCardanoEra where
                    => CardanoEra era    -- and explicit value.
                    -> AnyCardanoEra
 
+instance NtcVersionOf AnyCardanoEra where
+  ntcVersionOf (AnyCardanoEra _) = NodeToClientV_9
```

Comment:
Why are we defaulting to `NodeToClientV_9` for any era? Shouldn't we be explicit with each era and assign the version as follows:
```
--
data NodeToClientVersion
    = NodeToClientV_9
    -- ^ enabled @CardanoNodeToClientVersion7@, i.e., Alonzo
    | NodeToClientV_10
    -- ^ added 'GetChainBlockNo' and 'GetChainPoint' queries
    | NodeToClientV_11
    -- ^ added 'GetRewardInfoPools` Block query
    | NodeToClientV_12
    -- ^ added 'LocalTxMonitor' mini-protocol
    | NodeToClientV_13
    -- ^ enabled @CardanoNodeToClientVersion9@, i.e., Babbage
    | NodeToClientV_14
    -- ^ added @GetPoolDistr, @GetPoolState, @GetSnapshots
 ```

Reply:
I.e `NodeToClientV_12` for everything up to Alonzo and `NodeToClientV_14` for Babbage? The downside is we have to manually maintain this pairing.

Reply:
> Why are we defaulting to NodeToClientV_9 for any era?

`NodeToClientV_9` is the oldest era that is still being tracked.  Any queries that exist but are not in the above list above are presume to be from an era older than `NodeToClientV_9`, but we set it to `NodeToClientV_9` because it is long enough ago that we don't care about the precise version anymore.

> The downside is we have to manually maintain this pairing.

Yeah, I believe some form of this might belong in consensus.  It's implemented here for the moment because getting changes in consensus can take a long time.

Reply:
That's an interesting point.  Would it be sufficient for `nodeToClientVersionOf` to take an era argument?

Reply:
Note that we won't know the node's era unless we query for it, so it's not free.